providers/anthropic/anthropic.go

```
@@ -874,101 +958,80 @@ func forcedToolChoice(choice llm.ToolChoiceType) bool {
-		strings.HasPrefix(model, "claude-opus-4-8") ||
-		strings.HasPrefix(model, "claude-opus-5") ||
-		strings.HasPrefix(model, "claude-fable-5") ||
-		strings.HasPrefix(model, "claude-mythos-5") ||
-		strings.HasPrefix(model, "claude-sonnet-5")
+	caps, known := lookupCapabilities(model)
+	return known && !caps.manualBudget && caps.adaptive
+}
+
+// modelSupportsAdaptiveThinking reports whether thinking:{type:"adaptive"} is
+// accepted. The 4.5 generation predates it and answers 400.
+func modelSupportsAdaptiveThinking(model string) bool {
```

golangci-lint (2.12.2)

[error] func modelSupportsAdaptiveThinking is unused

(unused)